Add signed-metadata example

jprzychodzen opened the pull request:
* https://cloud.google.com/compute/docs/instances/verifying-instance-identity
  requires Java example

lesv commented on compute/signed-metadata/src/main/java/com/example/compute/signedmetadata/VerifyingInstance.java:
```diff
@@ -0,0 +1,65 @@
+      System.out.println("Project id : " + decodedJWT.getProjectId());
+      System.out.println("Project number : " + decodedJWT.getProjectNumber());
+      // This are examples how to handle exceptions, which indicate verification failure.
+    } catch (AlgorithmMismatchException e) {
```
very nice - Too few of our samples do this.

jprzychodzen replied:
Thanks!

lesv commented on compute/signed-metadata/pom.xml:
```diff
@@ -0,0 +1,66 @@
+  <artifactId>compute-signed-metadata</artifactId>
+  <version>1.0-SNAPSHOT</version>
+  <name>compute-signed-metadata</name>
+  <dependencies>
```
Usually we ask that you reference the parent pom .. so that checkstyle and other plugins run at verify time.

jprzychodzen replied:
Done

jprzychodzen commented:
I've added reference to parent POM.

This mostly integrates with GCP depended infrastructure. It is hard to verify tokens outside GCP - test would need to duplicate GCP token generation, have some kind of generated certificate and private/public key pair. And mostly be a standalone server on it's own.

Also, rebased branch to newest commit.

lesv commented:
Thank you - ping before your next submission and I'll add you to a team, so you can commit on a branch (that let's our testing work).

jprzychodzen commented:
Thanks for review and merging. I will ping before next submissions.
